### DISCUSSION TITLE
Qt application theming on GTK desktops

### DISCUSSION THREAD

**Discussion by banzaigtv** [SpiralLinux/SpiralLinux-project#411] [Q&A]
When testing and using various desktop spins, I found that the Budgie version is the only GTK spin which automatically forces Qt applications to inherit the active GTK desktop theme. It's one of the reasons why Budgie is likely the best version of SpiralLinux 12. I've seen where Linux Mint, Solus, and MX Linux do this for Cinnamon, Budgie, and XFCE, respectively. I've always had trouble with fixing this regardless of what tutorials I try to follow. Is there an easy way to implement this tactic on GNOME, Cinnamon, MATE, and XFCE? And could this fix be implemented in a future update?

**Comment by geckolinux** [comment]
> I found that the Budgie version is the only GTK spin which automatically forces Qt applications to inherit the active GTK desktop theme

Hi there, all SpiralLinux editions include the same feature. However, it should be noted that the feature only works with Qt5 apps, not Qt6. Also, in the case of Qt apps packaged as Flatpaks the theme can't be inherited automatically, although a similar theme to the SpiralLinux defaults is manually forced for Qt Flatpak apps.

**Comment by banzaigtv** [comment]
OK, so I tested the Cinnamon and XFCE versions in a VM and it seems to work. But I once installed the Cinnamon version on an external SSD and the fix did not seem to apply. Maybe I did something wrong? This time, I made sure to add custom themes to a hidden themes folder in the home directory. I'm going to try the GNOME version next and see if the Adwaita Dark theme applies.

EDIT: This does not work on GNOME automatically, but setting the Kvantum theme can manually set a Qt theme.

**Reply by geckolinux** [reply]
Hi, it has been tested to work on all editions. If you look at the screenshots on https://spirallinux.github.io you can see that VLC, which is a Qt5 app, is themed the same as the other GTK apps.

Which Qt app are you testing with, and how was it installed?

Also, how did you install the themes you're testing?

**Reply by banzaigtv** [reply]
I installed Palapeli, which is a KDE game application (Qt5), from the Bookworm repos.

**Reply by geckolinux** [reply]
Thanks, and how did you install the themes you're testing?

**Reply by banzaigtv** [reply]
I installed the custom themes on the .themes folder I created in the home directory. This works on all other desktops, but does not automatically pick up the GTK theme on GNOME. Perhaps GTK4 Libadwaita theming is to blame for this?

**Reply by geckolinux** [reply]
Do you have the URL where you downloaded the theme from?

**Reply by banzaigtv** [reply]
GNOME vs XFCE. GNOME with a pre-installed Yaru dark theme and XFCE with a custom dark theme.
![SpiralLinux GNOME VLC](https://github.com/SpiralLinux/SpiralLinux-project/assets/80161570/962e18f7-ce3b-4d73-bf6a-51a7bbc1915c)
![SpiralLinux XFCE VLC](https://github.com/SpiralLinux/SpiralLinux-project/assets/80161570/36d43ba0-73a0-45cb-98f9-f53376cb8282)

**Reply by geckolinux** [reply]
For the Gnome edition the explanation is documented in `/etc/environment` :

```
# This used to be set to `gtk2`, now using `kvantum` to avoid skinny Qt scrollbars with Yaru
QT_STYLE_OVERRIDE=kvantum

**Reply by banzaigtv** [reply]
Switched to GNOME's default Adwaita theme (dark variant) and VLC still looks like crap. And why is the accent color on the file manager still red? That does not become an issue on the Nemo file manager after it is installed.
![Screenshot_20240316_095823](https://github.com/SpiralLinux/SpiralLinux-project/assets/80161570/de651805-7f28-4f14-a880-e2ed97a44ae2)

**Comment by geckolinux** [comment]
In `/etc/environment` you would need to set `QT_STYLE_OVERRIDE=gtk2` in the case of the Gnome edition to make Qt5 apps imitate the active GTK theme.

The accent color in GTK4 apps is defined in `~/.config/gtk-4.0/gtk.css` .
